source/material_model/ascii_reference_profile.cc

```diff
@@ -0,0 +1,212 @@
+                             "A list of prefactors for the viscosity for each phase. The reference "
+                             "viscosity will be multiplied by this factor to get the corresponding "
+                             "viscosity for each phase. "
+                             "List must have one more entry than Phase transition depths. "
+                             "Units: non-dimensional.");
+        }
+        prm.leave_subsection();
+
+        aspect::Utilities::AsciiDataProfile<dim>::declare_parameters(prm,
+                                                                     "$ASPECT_SOURCE_DIR/data/adiabatic-conditions/ascii-data/",
+                                                                     "simple_test.txt");
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    AsciiReferenceProfile<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Ascii reference profile");
+        {
+          tala                 = prm.get_bool ("Use TALA");
+          thermal_conductivity = prm.get_double ("Thermal conductivity");
+          viscosity            = prm.get_double ("Viscosity");
+          thermal_viscosity_exponent = prm.get_double ("Thermal viscosity exponent");
+          transition_depths    = Utilities::string_to_double
+                                 (Utilities::split_string_list(prm.get ("Transition depths")));
+          viscosity_prefactors = Utilities::string_to_double
+                                 (Utilities::split_string_list(prm.get ("Viscosity prefactors")));
+
+          // make sure to check against the depth lists for size errors, since using depth
+          if (viscosity_prefactors.size() != transition_depths.size()+1)
+            AssertThrow(false, ExcMessage("Error: The list of Viscosity prefactors needs to have exactly "
+                                          "one more entry than the list of Transition depths. "));
+        }
+        prm.leave_subsection();
+
+        profile.parse_parameters(prm);
+      }
+      prm.leave_subsection();
+
+      // Declare dependencies on solution variables
+      this->model_dependence.viscosity = NonlinearDependence::none;
+      this->model_dependence.density = NonlinearDependence::pressure | NonlinearDependence::temperature;
+      this->model_dependence.compressibility = NonlinearDependence::none;
+      this->model_dependence.specific_heat = NonlinearDependence::none;
+      this->model_dependence.thermal_conductivity = NonlinearDependence::none;
+
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(AsciiReferenceProfile,
+                                   "ascii reference profile",
+                                   "A material model that reads in a reference "
+                                   "state from a text file.")
```

Comment:
I think this description should include (i) a discussion of the file format, (ii) a representation of the formulas that you use in the `evaluate()` function so that one can match the parameters to the actual model.